homeassistant/helpers/entity_registry.py

```diff
@@ -145,14 +144,21 @@ def async_generate_entity_id(
 
         Conflicts checked against registered and currently existing entities.
         """
-        return ensure_unique_string(
-            f"{domain}.{slugify(suggested_object_id)}",
-            chain(
-                self.entities.keys(),
-                self.hass.states.async_entity_ids(domain),
-                known_object_ids if known_object_ids else [],
-            ),
-        )
+        preferred_string = f"{domain}.{slugify(suggested_object_id)}"
+        test_string = preferred_string
+        if not known_object_ids:
+            known_object_ids = {}
+
+        tries = 1
+        while (
+            test_string in self.entities
+            or test_string in known_object_ids
+            or self.hass.states.get(test_string)
```

Comment:
Is it faster to do the state lookup here, than creating a set of all entity_ids first and then putting that set in this check on this line?

There is three function calls in the `self.hass.states.get(test_string)` lookup on this line, that's why I'm interested in this line.

Reply:
Most of the time was spent calling `self.hass.states.async_entity_ids(domain)` previously.

Reply:
https://koston.org/async_generate_entity_id_callgrind.tar.gz 
Github wouldn't let me upload the callgrinds so I put them here.

Reply:
Ok. Good to know!

Reply:
I think the callgrind files in the gzip archive had their names swapped. I'm a novice callgrinder but that's what it looks like to me.

Thanks for sharing this! Very useful tool!

Reply:
I definitely reversed them when I renamed them from callgrind.out.nnnnn